### DISCUSSION TITLE
ライブ配信が始まった際にローカルレコードが始まっているかを確認する方法はありますか？

### DISCUSSION THREAD

**Discussion by yuya-h-29** [HaishinKit/HaishinKit.swift#1044] [Q&A]
質問になります。#412  を参考にしてローカルレコードの機能を実装し、ほぼ毎回ローカルレコードができるようになりました。
ただ、`didFinishWriting`内の`performChanges(_:completionHandler:)`で時々以下の3305エラーが返ってきます。
エラーが出るときは毎回[.notEnoughSpace](https://developer.apple.com/documentation/photokit/phphotoserror/code/notenoughspace)が返ります。
<img width="642" alt="Screen Shot 2022-08-09 at 10 46 15" src="https://user-images.githubusercontent.com/54800510/183545287-ca5ce18e-6b48-4c46-a9a0-746a5bda84a2.png">

ローカルレコードで配信を始める前にiPhoneの残りストレージを出しているのですが、残りストレージが30GB以上あって2分くらいの配信のローカル保存でも失敗することもまれにありました。なので、そもそもストレージの問題ではないのかもしれないと考えています。
配信を開始した際にローカルレコードが成功しているか分かる方法はありますか？

### コード

```
class RecorderDelegate: DefaultAVRecorderDelegate {
    override func didFinishWriting(_ recorder: AVRecorder) {
        guard let writer: AVAssetWriter = recorder.writer else { return }
        PHPhotoLibrary.shared().performChanges({() -> Void in
            PHAssetChangeRequest.creationRequestForAssetFromVideo(atFileURL: writer.outputURL)
        }, completionHandler: {saved, error -> Void in
            var resultDic = Dictionary<String, String>()
            let title = "title"
            let message = "message"
            
            if let error = error {
                resultDic[title] = Alert.Title.error
                resultDic[message] = "動画の保存ができませんでした。: \(String(describing: error.localizedDescription))"
            } else if saved {
                resultDic[title] = "保存完了"
                resultDic[message] = "撮影したライブイベントは写真アプリに保存されました。"
            } else {
                resultDic[title] = "保存失敗"
                resultDic[message] = "動画の保存に失敗しました。"
            }
            NotificationCenter.default.post(name: .saveVideoToPhotoLibrary, object: nil, userInfo: resultDic)
            
            do {
                try FileManager.default.removeItem(at: writer.outputURL)
            } catch {
                print("error in recorder delegate: \(error.localizedDescription)")
            }
        })
    }
}
```

### Expected behavior
- ローカルレコード有りで配信が開始した際にローカルレコードができていない状態であれば`didFinishWriting`のタイミングでなく、　`didStartRunning`くらいのタイミングで失敗していることを通知したい（可能であれば）
- ローカル保存に失敗した際に動画ファイルを再度ローカル保存を試みたいのですが、別の解決策などありそうでしょうか？

### Smartphone (please complete the following information):
- iPhone SE2やiPhone13 pro
- OSは確認できていません。すみません。

**Comment by shogo4405** [comment]
- 1.2.x系では用意しておりません。一方で、秋頃リリース予定の1.3.0系では用意しており。エラーのハンドリングしやすいように仕様変更しております。 
    - https://github.com/shogo4405/HaishinKit.swift/blob/main/Sources/Media/AVRecorder.swift#L6

**Reply by yuya-h-29** [reply]
@shogo4405 
返信遅れてしまいすみません。
仕様変更に関して承知しました！いつも有難うございます！！
